cdap-api-common/src/main/java/co/cask/cdap/api/data/schema/Schema.java

```diff
@@ -99,21 +99,53 @@ public boolean isSimpleType() {
    *
    */
   public enum LogicalType {
-    DATE(Type.INT),
-    TIMESTAMP_MILLIS(Type.LONG),
-    TIMESTAMP_MICROS(Type.LONG),
-    TIME_MILLIS(Type.INT),
-    TIME_MICROS(Type.LONG);
+    DATE(Type.INT, "date"),
+    TIMESTAMP_MILLIS(Type.LONG, "timestamp-millis"),
+    TIMESTAMP_MICROS(Type.LONG, "timestamp-micros"),
+    TIME_MILLIS(Type.INT, "time-millis"),
+    TIME_MICROS(Type.LONG, "time-micros");
 
     private final Type type;
+    private final String name;
+
+    private static final Map<String, LogicalType> LOOKUP_BY_NAME;
+    static {
+      LOOKUP_BY_NAME = new HashMap<>();
```

Comment:
should make sure this map can't be changed. Create a HashMap and wrap it with Collections.unmodifiableMap() when assigning to LOOKUP_BY_NAME at the end of the static block.